### PR TITLE
refactor: Search results container

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Refactor**

### Description
A small refractor to improve the readability of the code for `Search` component

### PR Checklist (tick all before merging)

- [ ] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [x] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Split search results from Search component - dimatretyak

</details>

### CODE REVIEW

File: src/app/Scenes/Search/Search.tsx [hunk 1]
```diff
@@ -310,7 +220,11 @@ export const Search: React.FC = () => {
                     isSelected={isSelected}
                   />
                 </Box>
-                {renderResults()}
+                <SearchResults
+                  selectedPill={selectedPill}
+                  query={searchQuery}
+                  onRentry={handleRetry}
```
Comment (gkartalis) [inline src/app/Scenes/Search/Search.tsx:226]: small typo here
```suggestion
                  onRetry={handleRetry}
```

File: src/app/Scenes/Search/SearchResults.tsx [hunk 2]
```diff
@@ -1,123 +1,105 @@
...
 
-interface SearchResultsProps
-  extends StateResultsProvided<AlgoliaSearchResult>,
-    InfiniteHitsProvided<AlgoliaSearchResult> {
+interface SearchResultsProps {
   selectedPill: PillType
-  onRetry?: () => void
-  trackResultPress?: (result: AlgoliaSearchResult) => void
+  query: string
+  onRentry: () => void
```
Comment (gkartalis) [inline src/app/Scenes/Search/SearchResults.tsx:16]:
```suggestion
  onRetry: () => void
```

**Review by gkartalis — DISMISSED** [review]
🚢  One small typo detected! Other than that looking great! 

Thanks for this refactor 🔥

File: src/app/Scenes/Search/Search.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/app/Scenes/Search/Search.tsx:226]: good catch 🕵️‍♂️

Comment (dimatretyak) [reply src/app/Scenes/Search/Search.tsx]: Fixed

**Review by gkartalis — APPROVED** [review]

**Pull request merged**